cpp-client/deephaven/client/src/client_options.cc

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+ */
+#include "deephaven/client/client_options.h"
+#include "deephaven/dhcore/utility/utility.h"
+
+using deephaven::dhcore::utility::base64Encode;
+
+namespace deephaven::client {
+
+ClientOptions::ClientOptions() {
+  setDefaultAuthentication();
+  setSessionType("python");
+}
+
+ClientOptions::ClientOptions(ClientOptions &&other) noexcept = default;
+ClientOptions &ClientOptions::operator=(ClientOptions &&other) noexcept = default;
+ClientOptions::~ClientOptions() = default;
+
+ClientOptions &ClientOptions::setDefaultAuthentication() {
+  authorizationValue_ = "Anonymous";
+  return *this;
+}
+
+ClientOptions &ClientOptions::setBasicAuthentication(const std::string &username, const std::string &password) {
+  auto token = username + ':' + password;
+  authorizationValue_ = "Basic " + base64Encode(token);
+  return *this;
+}
+
+ClientOptions &ClientOptions::setCustomAuthentication(const std::string &authenticationType,
+    const std::string &authenticationToken) {
+  authorizationValue_ = authenticationType + " " + authenticationToken;
+  return *this;
+}
+
+ClientOptions &ClientOptions::setSessionType(const std::string &sessionType) {
+  this->sessionType_ = sessionType;
+  return *this;
+}
+
+ClientOptions &ClientOptions::setUseTls(const bool useTls) {
+  useTls_ = useTls;
+  return *this;
+}
+
+ClientOptions &ClientOptions::setTlsRootCerts(const std::string tlsRootCerts) {
+  tlsRootCerts_ = std::move(tlsRootCerts);
+  return *this;
+}
+
+ClientOptions &ClientOptions::setClientCertChain(const std::string clientCertChain) {
+  clientCertChain_ = std::move(clientCertChain);
+  return *this;
+}
+
+ClientOptions &ClientOptions::setClientPrivateKey(const std::string clientPrivateKey) {
+  clientPrivateKey_ = std::move(clientPrivateKey);
+  return *this;
+}
+
+ClientOptions &ClientOptions::addIntOption(const std::string opt, const int val) {
+  intOptions_.emplace_back(std::move(opt), val);
+  return *this;
+}
+
+ClientOptions &ClientOptions::addStringOption(const std::string opt, const std::string val) {
```

Comment:
Please remove `const` here, because `const` is preventing std::move from happening and downgrading the operation to a copy.